src/nextcloud/bin/nextcloud-fixer

```diff
@@ -24,6 +24,12 @@ if nextcloud_is_installed; then
 	trap 'disable_maintenance_mode' EXIT
 
 	occ -n db:convert-filecache-bigint
+
+	echo "Updating all apps..."
+	if occ -n app:update --all; then
+		# No idea why, but Nextcloud wants an upgrade right after
+		occ -n upgrade
```

Comment:
@kesselb any idea why this is required?

Reply:
Yes. occ app:update download the app updates and extract the code to the right place. Next step is to run occ upgrade routine (migrations, bump app version). I don't know why. Probably a "design" design in the past. The same process for updates via the official updater. Updater will fetch the latest release, put the code in place, run occ upgrade routine.

Reply:
After running snap refresh and occ upgrade all apps should be up to date (occ upgrade will check the appstore for updates). But https://github.com/nextcloud/server/blob/master/core/Command/App/Update.php will also return 0 if nothing happens.

Reply:
Ah this is by design then. Seems a little odd, but that makes me feel better! It's okay if the update step returns zero, the upgrade step also returns zero if it's already up to date (despite a comment I have to the contrary, so I think that has been different in the past).

Reply:
I've updated the comments here to reflect reality as I understand it, thank you for the explanation.